### ISSUE TITLE
Using 'set -o pipefail' on grep commands causes fatal errors when no matches are found.

### ISSUE THREAD

**Issue by kresss** [ComplianceAsCode/content#4783]
#### Description of problem:
When running the ansible STIG playbook a fatal error is encountered for `TASK [Get nfs and nfs4 mount points, that don't have Kerberos security option]`.

I think this is is caused by having no nfs mounts. The task includes a shell command containing multiple grep command piped together with 'set -o pipefail' enabled. Grep returns a non-zero value when no results are returned. Removing the pipefail option allows the task to succeed. 

#### SCAP Security Guide Version:
0.1.45

#### Operating System Version:
```
[kresss@kress ansible]$ cat /etc/redhat-release 
Red Hat Enterprise Linux Server release 7.7 (Maipo)
[kresss@kress ansible]$ 
[kresss@kress ansible]$ ansible --version
ansible 2.8.3
  config file = /home/kresss/GIT/MCS/ansible/ansible.cfg
  configured module search path = [u'/home/kresss/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 14:33:56) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

#### Steps to Reproduce:

1. On a system with no nfs mounts in /etc/fstab
2. ansible-playbook -i "localhost," -c local -C rhel7-playbook-stig.yml

#### Actual Results:
```
[kresss@kress ansible]$ ansible-playbook -i "localhost," -c local -C rhel7-playbook-stig.yml

PLAY [all] ********************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Disable service kdump] **************************************************************************************************************************************************************************************
ok: [localhost]

TASK [Disable socket of service kdump if applicable] **************************************************************************************************************************************************************
changed: [localhost]

TASK [Test for existence /etc/cron.allow] *************************************************************************************************************************************************************************
ok: [localhost]

TASK [Ensure group owner 0 on /etc/cron.allow] ********************************************************************************************************************************************************************
skipping: [localhost]

TASK [Test for existence /etc/cron.allow] *************************************************************************************************************************************************************************
ok: [localhost]

TASK [Ensure owner 0 on /etc/cron.allow] **************************************************************************************************************************************************************************
skipping: [localhost]

TASK [Ensure vsftpd is removed] ***********************************************************************************************************************************************************************************
ok: [localhost]

TASK [Get nfs and nfs4 mount points, that don't have Kerberos security option] ************************************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "cmd": "set -o pipefail\ngrep -E \"[[:space:]]nfs[4]?[[:space:]]\" /etc/fstab | grep -v \"sec=krb5:krb5i:krb5p\" | awk '{print $2}'\n", "delta": "0:00:00.007669", "end": "2019-08-30 16:37:49.351222", "msg": "non-zero return code", "rc": 1, "start": "2019-08-30 16:37:49.343553", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=6    changed=1    unreachable=0    failed=1    skipped=2    rescued=0    ignored=0   

```

#### Expected Results:
The playbook should run.

#### Addition Information/Debugging Steps:

If it is acceptable to not use the pipefail option, I think the following two files need changed:

1. shared/templates/template_ANSIBLE_mount_option_remote_filesystems
2. content/linux_os/guide/services/nfs_and_rpc/nfs_configuring_clients/mounting_remote_filesystems/mount_option_krb_sec_remote_filesystems/ansible/shared.yml

If removing the pipefail option is not an option, I would suggest doing the whole thing in single awk command.

**Comment by redhatrises** [comment]
@kresss Thanks for the PR!

Looking at the ansible remediations.... they should probably just get a re-write. The tasks are a bit hokey for how they do things.

**Comment by kresss** [comment]
@redhatrises Thanks for the quick reply. I didn't create a PR. If you want one I can remove the problem `set -o pipefail` lines, but I am not the person you want to do a full task re-write.

Looks like some of the rpm -Va rules have/had this problem too. The 0.1.45 release includes a similar bug for those checks, but @mildas already fixed those issues in 3be27a167efbcbec086913c5e35d5fba5aa6f2b8.

**Comment by redhatrises** [comment]
> @redhatrises Thanks for the quick reply. I didn't create a PR. If you want one I can remove the problem `set -o pipefail` lines, but I am not the person you want to do a full task re-write.
> 
> Looks like some of the rpm -Va rules have/had this problem too. The 0.1.45 release includes a similar bug for those checks, but @mildas already fixed those issues in [3be27a1](https://github.com/ComplianceAsCode/content/commit/3be27a167efbcbec086913c5e35d5fba5aa6f2b8).

@kresss no worries.

The PR is being reverted because it isn't the correct fix.
